src/provider/types.ts

```diff
@@ -0,0 +1,130 @@
+export type FastlyAccount = {
```

Comment:
tldr; nothing needs to change here, but for the sake of driving integrations to consistency...

I think the properties on these types will not be maintained since most of them are not referenced anywhere in the code. The only value of TypeScript typings is at development time, so where they're not used, they're not worth maintaining IMO.

If the goal is to reveal where the provider API changes what it sends back, that will need to be done with something like https://github.com/pelotom/runtypes, which can validate the API is what we expect at runtime, and when the API changes unexpectedly, the runtypes will throw an error so we can address the problem.

If the goal is to document what comes from the API, the recordings and snapshots serve well. A developer can figure what needs special conversion for assign properties that are not handled by `...convertProperties()` by looking at the API recordings. Also, the snapshots are the only place to see what is actually being sent to the persister since there are not likely to be tests for results of `...convertProperties()`.

Of course, if no effort is made to record or assert against snapshots...